deps: update the Nomad API dependency to v1.7.2

jrasell opened the pull request:
### What does this PR do?
Update the Nomad API dependency to v1.7.2.

### Motivation
Along with some minor fixes, this new version of the Nomad API package adds support for connectivity via UNIX domain sockets. When Traefik is being run as a Nomad job, this change will allow operators to use the socket that Nomad automatically injects into the task namespace for API connectivity.

I have targeted the v3.0 branch as I don't feel this modification falls under the bug fix category, as detailed in your contributing guide. Please let me know if there are any questions or problems.

nmengin commented:
Hello @jrasell,

Thank you for your contribution.
After discussing it with other maintainers, we think that your PR can be brought in Traefik v2.11.
To do so, could you rebase your branch on `v2.11`?

jrasell commented:
> Hello @jrasell,
> 
> Thank you for your contribution. After discussing it with other maintainers, we think that your PR can be brought in Traefik v2.11. To do so, could you rebase your branch on `v2.11`?

Hi @nmengin and thanks for taking a look. I have updated the base branch to `v2.11` as requested and modified the PR target branch to match.

kevinpollet commented:
Hello @jrasell,

Could you fix the conflicts?

jrasell commented:
> Hello @jrasell,
> 
> Could you fix the conflicts?

Hi @kevinpollet, those have now been fixed.

kevinpollet reviewed (approved):
Thanks 👍

nmengin reviewed (approved):
LGTM, thanks @jrasell 👍

ldez reviewed (approved):
LGTM
